src/Accounts/Authentication.ResourceManager/AzureRmProfile.cs

```diff
@@ -188,7 +205,7 @@ private void Initialize(AzureRmProfile profile)
                     this.Contexts.Add(context.Key, context.Value);
                 }
 
-                DefaultContextKey = profile.DefaultContextKey ?? "Default";
+                DefaultContextKey = profile.DefaultContextKey ?? (profile.Contexts.Any() ? null : "Default");
```

Comment:
Setting the defautl key to null if there are contexts but none selected - when would this occur?

Reply:
Needs validate: If logged in in VS (not PS), when first time runing PS command (not select-context), should be prompted with "please select a context"